### PR TITLE
java: allow user to provide custom channels.

### PR DESCRIPTION
**Pull request by zklapow**
At HubSpot we need to install a custom EventLoop for channels in vitess. In the past, we have done this simply by overriding the event loop for our internal build. This PR provides a more extensible way for the user to provide customized channels to the vitess client.

The existing `RetryingInterceptor` logic has been moved to the new `DefaultChannelBuilderProvider`

@leoxlin @acharis FYI

### CODE REVIEW

**Comment by sougou** [comment]
@mpawliszyn @leoxlin please review.

File: java/grpc-client/src/main/java/io/vitess/client/grpc/netty/DefaultChannelBuilderProvider.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import io.grpc.netty.NettyChannelBuilder;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.util.concurrent.DefaultThreadFactory;
+import io.vitess.client.grpc.RetryingInterceptor;
+import io.vitess.client.grpc.RetryingInterceptorConfig;
+
+public class DefaultChannelBuilderProvider implements NettyChannelBuilderProvider {
+  private static final EventLoopGroup ELG = new NioEventLoopGroup(
```
Comment (leoxlin) [inline java/grpc-client/src/main/java/io/vitess/client/grpc/netty/DefaultChannelBuilderProvider.java]: We should probably exclude the ELG change in the default provider as this deviates from former behavior

File: java/jdbc/src/main/java/io/vitess/jdbc/VitessVTGateManager.java [hunk 2]
```diff
@@ -222,6 +231,35 @@ private static RetryingInterceptorConfig getRetryingInterceptorConfig(VitessConn
         conn.getGrpcRetryMaxBackoffMillis(), conn.getGrpcRetryBackoffMultiplier());
   }
 
+  private static NettyChannelBuilderProvider getChannelProviderFromProperties(
```
Comment (leoxlin) [inline java/jdbc/src/main/java/io/vitess/jdbc/VitessVTGateManager.java]: Can we add an unit test for this?

**Comment by leoxlin** [comment]
LGTM once the build failures are resolved.

**Comment by sougou** [comment]
@zklapow ping

**Comment by morgo** [comment]
@zklapow I am going to close this PR due to staleness, but please feel free to re-open if you would like to bring it back to life. Thanks!

**Pull request closed without merging**